File: chap-1/refactored/javascript/tragedyCalculator.js
const PerformaceCalculator = require('./performaceCalculator');

module.exports = class TragedyCalculator extends PerformaceCalculator {

    get amount() {
        let result = 40000
        if (this.performance.audience > 30) {
            result += 1000 * (this.performance.audience - 30)
        }
        return result;
    }
}

File: chap-1/refactored/java/src/main/java/calculator/TragedyCalculator.java
package calculator;

import data.invoice.Performance;

public class TragedyCalculator extends PerformaceCalculator {

    public TragedyCalculator(Performance performance) {
        super(performance);
    }

    @Override
    public double amount() {
        double result = 40000;
        if (this.performance.getAudience() > 30) {
            result += 1000 * (this.performance.getAudience() - 30);
        }
        return result;
    }
}
